mleap-xgboost-runtime/src/test/scala/ml/combust/mleap/xgboost/runtime/XGBoostClassificationModelParitySpec.scala

```diff
@@ -0,0 +1,128 @@
+
+        val singleRowDMatrix = r(1).asInstanceOf[SparseTensor[Double]].asXGB
+
+        val boosterResult = booster.predict(singleRowDMatrix, false, 0).head(0)
+
+        val boosterProbability = Vectors.dense(1 - boosterResult, boosterResult).toDense
+        val boosterPrediction = Math.round(boosterResult)
+
+
+        assert (boosterPrediction == mleapResult.dataset.head.getDouble(mleapPredictionColIndex))
+
+        assert (
+          almostEqualSequences(
+            Seq(boosterProbability.values),
+            Seq(mleapResult.dataset.head.getTensor[Double](mleapProbabilityColIndex).toArray)
+          )
+        )
+
+        val boosterResultWithMargin = booster.predict(singleRowDMatrix, true, 0).head(0)
+        val boosterRawPrediction = Vectors.dense(- boosterResultWithMargin, boosterResultWithMargin).toDense
+
+        assert (almostEqualSequences(
+          Seq(boosterRawPrediction.values),
+          Seq(mleapResult.dataset.head.getTensor[Double](mleapRawPredictionColIndex).toArray)
+        ))
+    }
+  }
+
+  it("Results between the XGBoost4j booster and the MLeap Transformer are the same") {
+    val booster = trainBooster(xgboostParams, denseDataset)
+    val xgboostTransformer = createBoosterClassifier(trainBooster(xgboostParams, denseDataset))
+
+    val mleapBundle = serializeModelToMleapBundle(xgboostTransformer)
+    val deserializedTransformer: Transformer = loadMleapTransformerFromBundle(mleapBundle)
+
+    equalityTestRowByRow(booster, deserializedTransformer)
+  }
+
+  it("has the correct inputs and outputs with columns: prediction, probability and raw_prediction") {
+
+    val booster = trainBooster(xgboostParams, denseDataset)
+    val transformer = createBoosterClassifier(booster)
+
+    assert(transformer.schema.fields ==
+      Seq(StructField("features", TensorType(BasicType.Double, Seq(numFeatures))),
+        StructField("raw_prediction", TensorType(BasicType.Double, Seq(2))),
+        StructField("probability", TensorType(BasicType.Double, Seq(2))),
+        StructField("prediction", ScalarType.Double.nonNullable)))
+  }
+
+  it("Results are the same pre and post serialization") {
+    val booster = trainBooster(xgboostParams, denseDataset)
+    val xgboostTransformer = createBoosterClassifier(booster)
+
+    val preSerializationResult = xgboostTransformer.transform(leapFrameLibSVMtrain)
+
+    val mleapBundle = serializeModelToMleapBundle(xgboostTransformer)
+
+    val deserializedTransformer: Transformer = loadMleapTransformerFromBundle(mleapBundle)
+    val deserializedModelResult = deserializedTransformer.transform(leapFrameLibSVMtrain).get
+
+    assert(preSerializationResult.get.dataset == deserializedModelResult.dataset)
+  }
+
+  it("Test XGBoostMultinomialClassificationModel results are the same pre and post serialization") {
+    val booster = trainBooster(xgboostParams, denseDataset)
+    val xgboostTransformer = createMultinomialBoosterClassifier(booster)
```

Comment:
Is there a test for `MultinomialBoosterClassifier` parity with XGBoost4J anywhere?

Reply:
I added one using a mutli-label Iris dataset